### DISCUSSION TITLE
I ran this with Qt.

### DISCUSSION THREAD

**Discussion by MythicalForce** [textbrowser/glitch#3] [Q&A]
is everything you can do at this point just opning the program? i can`t do anything inside (like code stuff)!

**Comment by textbrowser** [comment]
Here's a brief video of editing.

https://user-images.githubusercontent.com/10701156/155199940-bbdac257-6397-48d8-90ec-0dfb71c2f746.mp4

**Comment by textbrowser** [comment]
You can drag-and-drop completed objects. I'm now working on wiring objects. For example, the output of a constant can be wired as an input to a function.

Remaining beyond the TO-DO: generate Java from a diagram (the compiler), wiring, and all of the Arduino objects. So there's a lot left. :)

**Comment by textbrowser** [comment]
Thanks for showing an interest. :)

**Reply by MythicalForce** [reply]
nooo problemo i have been searching for YEARS! after something like this!! i have always been a fan of the "blueprint system"! that Epic Games uses in the Unreal Engine!

**Reply by textbrowser** [reply]
You'll find out when it's done. It may also work on Android. :)

The basic idea: design -> generate the Java source (compiler) -> use the native Arduino tool to build the generated Java and upload the results to the computer.

**Comment by textbrowser** [comment]
Now generating source from diagrams. Please see images. I'm progressing through the Arduino examples. For each example, I create the diagram and then generate the source.

https://textbrowser.github.io/glitch/
